src/backend/distributed/commands/role.c

```diff
@@ -88,6 +90,55 @@ PostprocessAlterRoleStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PreprocessAlterRoleSetStmt actually creates the plan we need to execute for alter
+ * role set statement.
+ */
+List *
+PreprocessAlterRoleSetStmt(Node *node, const char *queryString)
+{
+	if (!EnableAlterRolePropagation)
+	{
+		return NIL;
+	}
+
+	AlterRoleSetStmt *stmt = castNode(AlterRoleSetStmt, node);
+	ErrorIfUnsupportedAlterRoleSetStmt(stmt);
+
+	EnsureCoordinator();
+	QualifyTreeNode((Node *) stmt);
+	const char *sql = DeparseTreeNode((Node *) stmt);
+
+	List *commandList = list_make1((void *) sql);
+
+	return NodeDDLTaskList(ALL_WORKERS, commandList);
+}
+
+
+/*
+ * ErrorIfUnsupportedAlterRoleSetStmt raises an error if the AlterRoleSetStmt contains a
+ * construct that is not supported.
+ *
+ * Unsupported Constructs:
+ *  - ALTER ROLE ... SET ... FROM CURRENT
+ */
+static void
+ErrorIfUnsupportedAlterRoleSetStmt(AlterRoleSetStmt *stmt)
+{
+	VariableSetStmt *setStmt = stmt->setstmt;
+
+	if (setStmt->kind == VAR_SET_CURRENT)
+	{
+		/* check if the set action is a SET ... FROM CURRENT */
+		ereport(ERROR, (errmsg("unsupported ALTER ROLE ... SET ... FROM "
```

Comment:
We would be regressing from PostgreSQL here. Could we change this to a WARNING and not propagate it?

Reply:
I realized that we use `NOTICE` log level similar logs. I changed it appropriately.

I also added some notices where a user changes defaults for a different database, and we do not propagate it